analyzers/src/SonarAnalyzer.Common/Rules/SymbolicExecutionRunnerBase.cs

```diff
@@ -103,27 +108,46 @@ private void AnalyzeRoslyn(SonarAnalysisContext sonarContext, SyntaxNodeAnalysis
 
         protected class RuleFactory
         {
-            public Type Type { get; }
             private readonly Func<SymbolicRuleCheck> createInstance;
+            private readonly Func<object> sonarFallbackFactory;
+
+            public Type Type { get; }
 
-            protected RuleFactory(Type type, Func<SymbolicRuleCheck> createInstance)
+            protected RuleFactory(Type type, Func<SymbolicRuleCheck> createInstance, Func<object> sonarFallbackFactory)
             {
                 Type = type;
                 this.createInstance = createInstance;
+                this.sonarFallbackFactory = sonarFallbackFactory;
             }
 
-            public SymbolicRuleCheck CreateInstance(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext)
+            public SymbolicRuleCheck CreateInstance(IAnalyzerConfiguration configuration, SonarAnalysisContext sonarContext, SyntaxNodeAnalysisContext nodeContext)
             {
+                if (configuration.ForceSonarCfg && sonarFallbackFactory != null)
+                {
+                    return null;
+                }
+
                 var ret = createInstance();
                 ret.Init(sonarContext, nodeContext);
                 return ret;
             }
+
+            public object CreateSonarFallback(IAnalyzerConfiguration configuration) =>
+                configuration.ForceSonarCfg ? sonarFallbackFactory?.Invoke() : null;
```

Comment:
Same thing, just lowering the brain load to understand the logic. The null-conditional makes it a bit harder to follow the intention.
```suggestion
                configuration.ForceSonarCfg && sonarFallbackFactory is not null ? sonarFallbackFactory() : null;
```